### ISSUE TITLE
Limit heap size in JVM

### ISSUE THREAD

**Issue by rishabhpoddar** [supertokens/supertokens-core#241]
Add the -Xmx option that limits the maximum heap size for the jvm.

Quick overview: https://alvinalexander.com/blog/post/java/java-xmx-xms-memory-heap-size-control/

**Comment by C-mmon** [comment]
I would love to take this issue and work on it.
This will be very useful for devices such as raspbery pi, allowing better memory management.

**Comment by rishabhpoddar** [comment]
Hey @C-mmon. Thanks for showing interest. The idea here is to be able to start the core in a way that limits the memory consumption. 

The config to provide a ceiling value for memory can be provided in the config.yaml file, or in the CLI command line that starts the java process. What do you think?

**Comment by C-mmon** [comment]
Yes, I will try to determine range that will allow it to work perfectly. Like determining the initial value.
As well as document the whole thing, with correct notation.

**Comment by rishabhpoddar** [comment]
Sounds good! Thank you. I'll assign this issue to you.
